core/server/router_test.go

```diff
@@ -461,3 +463,69 @@ func TestRegisterErrorHandler(t *testing.T) {
 		}
 	})
 }
+
+// testHijackResponseWriter wraps the httptest.ResponseRecorder to implement Hijack interface.
+type testHijackResponseWriter struct {
```

Comment:
@tatang26 should this be part of the internal/writer package? I'm seeing that this test covers some of the functionallity in the writer so my first thought was it should live there.

Reply:
@paganotoni thank you for your thoughts!
My point in leaving tests live in the server package was because I wanted to check how the writer behaves in the current router flow. Taking care of other wrappers, such as the saver (which was modified to implement the writer), keeps the http implementations.

I agree with you about maybe adding tests to the writer package that check that the HTTP implementations of Flusher or Hijacker keep working when there are other response writer wrappers.